homeassistant/components/command_line/sensor.py

```diff
@@ -145,6 +145,8 @@ async def async_update(self) -> None:
                     None,
                 )
             )
+        elif self._json_attributes:
+            self._attr_native_value = None
```

Comment:
We already checked the condition of self._json_attribute so we can move this part under that condition.

Reply:
> We already checked the condition of self._json_attribute so we can move this part under that condition.

So, something like this at the end of the json_attributes block?

```
if self._value_template is None:
    self._attr_native_value = None
    return
```
 The `return` would be needed to prevent it from falling through to the next if/else block

Or maybe avoid that by

```
if self._value_template is None:
    value = None
```
So the default value then because None

Reply:
I've updated this to do it the second way, and have also updated the unit tests to ensure that the change works (when json_attributes are used, state is only set if value_template is also set).

Reply:
Looks good now I think